jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -483,9 +483,8 @@ destroyJVM (JVM_ jvm) = do
         (*$(JavaVM *jvm))->DestroyJavaVM($(JavaVM *jvm));
         jniEnv = NULL;
     } |]
-    cleanerMaybe <- readIORef globalRefCleaner
-    maybe (return ()) stopGlobalRefCleaner cleanerMaybe
-    writeIORef globalRefCleaner Nothing
+    stopGlobalRefCleaner <$> readIORef globalRefCleaner
```

Comment:
I think you want `readIORef globalRefCleaner >>= stopGlobalRefCleaner` here.

Reply:
Right, thanks.
fixed